src/main/scala/scorex/core/network/NetworkController.scala

```diff
@@ -242,14 +238,14 @@ class NetworkController(settings: NetworkSettings,
       // Drop connections with peers if they seem to be inactive
       val now = networkTime()
       connections.values.foreach { cp =>
-        val lastSeen = cp.peerInfo.map(_.lastSeen).getOrElse(now)
-        // A peer should send out sync message to us at least once per settings.syncStatusRefreshStable duration.
-        // We wait for more, namely settings.syncStatusRefreshStable.toMillis * 3
-        val timeout = settings.syncStatusRefreshStable.toMillis * 3
-        val delta = now - lastSeen
-        if (delta > timeout) {
-          log.info(s"Dropping connection with ${cp.peerInfo}, last seen ${delta / 1000.0} seconds ago")
-          cp.handlerRef ! CloseConnection
+        val lastSeen = cp.lastMessage
+        if (lastSeen != 0) {
```

Comment:
Then if no messages ever received from a peer and connection is dead, the connection will never be removed. I suggested to set lastMessage = now then creating connection, and then the check is not needed.

Reply:
Done